packages/@okta/vuepress-site/docs/api/resources/oidc/index.md

```diff
@@ -1332,33 +1202,72 @@ For more information about configuring an app for OpenID Connect, including grou
 * The full set of claims for the requested scopes is available via the [/oauth2/v1/userinfo](#userinfo) endpoint. Call this endpoint using the access token.
 
 ### Refresh Token
-
 Refresh tokens are opaque. More information about using them can be found in the [Authentication Guide](/authentication-guide/tokens/refreshing-tokens).
 
-## Token Authentication Methods
+## Client Authentication Methods
+When registering an OAuth 2 client application, you can specify an authentication method by including the [token_endpoint_auth_method](https://developer.okta.com/docs/api/resources/apps/#add-oauth-2-0-client-application) parameter. Okta supports five client authentication methods:
+
+> Note: If you don't specify a method when registering your client, the default method is `client_secret_basic`.
+
+* `client_secret_basic`, `client_secret_post`, `client_secret_jwt`: Use one of these three methods to authenticate against the `/token` or `/revoke` endpoints when the client has access to the `client_secret`. Typically services support client authentication via HTTP Basic Auth (`client_secret_basic`) with the client's `client_id` and `client_secret`. However, some services support authentication by accepting the `client_id` and `client_secret` as POST body parameters (`client_secret_post`). Check the service's documentation to find out what the service expects, since the OAuth 2.0 specification leaves this decision to the service.
```

Comment:
We shouldn't specify the endpoints along with the authentication methods, any client authentication method can be used against any endpoint which requires client authentication. Additionally the `/revoke` endpoint requires client authentication.

Reply:
Also Okta is the service in this situation and we accept any of these methods. It's up to the client to decide which one is most appropriate for them to use based on what kind of client they are/what the client is able to provide. For instance a Native client cannot secure a secret so should not use any of these. In other situations you might be using a client which was built by some other party and it might only support Basic Auth, so you would want to need to use that client auth mechanism.